src/hyp_map.rs

```diff
@@ -254,6 +257,72 @@ impl UmodeElfRegion {
     }
 }
 
+/// The U-mode Input Region is a private region of the page table used by the hypervisor to pass
+/// data to the current U-mode operation. This avoids mapping hypervisor data directly in
+/// U-mode. This area is written by the hypervisor and mapped read-only in U-mode.
+struct UmodeInputRegion {
+    /// Volatile Slice used to write to this area from the hypervisor.
+    vslice: VolatileSlice<'static>,
+}
+
```

Comment:
nit: If you don't need any other members then you can use the alternative `struct UmodeInputRegion(VolatileSlice<'static>)` syntax and then refer to it as `.0`

Reply:
I just realised that this a rearrangment commit so you can disregard these comments as this is not new code (or consider fixing in another commit if you think they're appropriate :-) )

Reply:
Heh, yes. In general I don't mind named structs. I find them simpler. Do you have any opinion on why would be better to have in the format you suggest?

Reply:
I think it's just a style/convention thing. If the struct is really a wrapper around another type (so you can implement methods or traits) then I think the tuple format is preferred. If it's truly a new struct that only happens to have one member then named member structs are the norm.

Reply:
I think I'll leave it like this for now. In a way there's more to the UmodeInputRegion than a slice, although that's what it is in practice.